engine/engine.go

```diff
@@ -306,10 +306,17 @@ loop:
 					if len(series[s].Points) == 0 {
 						series[s].Points = make([]promql.Point, 0, 121) // Typically 1h of data.
 					}
-					series[s].Points = append(series[s].Points, promql.Point{
-						T: vector.T,
-						V: vector.Samples[i],
-					})
+					if i < len(vector.HistogramSamples) {
```

Comment:
Does this mean we will either have floats or histograms in the result, but not both?